src/main/java/seedu/address/logic/parser/Parser.java

```diff
@@ -127,56 +129,61 @@ public Command parseCommand(String userInput) {
 	}
 
 	/**
-     * Parses arguments in the context of the add Task command.
-     *
-     * @param args full command args string
-     * @return the prepared command
-     */
-    private Command prepareAdd(String args) {
-    	args=args.trim();
-        final Matcher matcher = PERSON_DATA_ARGS_FORMAT.matcher(args.trim());//Todos: replace with TASK_DATA_ARGS_FORMAT
-        // Validate arg string format
-        if (!matcher.matches()) {
-            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
-        }
-            String type = idTaskType(args);
-        	switch(type){
-        	case("float"):
-        		return new AddFloatingTaskParser().parse(args);
-        	
-        	case("deadline"):
-        		return new AddDeadlineParser().parse(args);
-        	
-        	case("event"):
-        		return new AddEventParser().parse(args);	
-        	}
-        	return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));	
-        }
-
-    /**
-     * Identifies the type of task.
-     * 
-     * @param args
-     * @return type of task as String
-     */
-	private String idTaskType(String args) {
-		int s = wordCount(args);
-		switch (s) {
-		case (1):
-			return "float";
+	 * Parses arguments in the context of the add Task command.
+	 *
+	 * @param args
+	 *            full command args string
+	 * @return the prepared command
+	 */
+	private Command prepareAdd(String args) {
+		args = args.trim();
+		final Matcher matcher = PERSON_DATA_ARGS_FORMAT.matcher(args.trim());// Todos:
+																				// replace
+																				// with
+																				// TASK_DATA_ARGS_FORMAT
+		// Validate arg string format
+		if (!matcher.matches()) {
+			return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
+		}
+		String type = idTaskType(args);
+		switch (type) {
+		case ("float"):
+			return new AddFloatingTaskParser().parse(args);
 
-		case (2):
-			return "float";
+		case ("deadline"):
+			return new AddDeadlineParser().parse(args);
 
-		case (3):
-			return "deadline";
+		case ("event"):
+			return new AddEventParser().parse(args);
+		}
+		return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
+	}
 
-		case (5):
-			return "event";
+	/**
+	 * Identifies the type of task.
+	 * 
+	 * @param args
+	 * @return type of task as String
+	 */
+	private String idTaskType(String args) {
+
+		// float, no date or time
+		// deadline, has date or/and time
+		// event has date and time and to keyword
+		String[] s = args.split(" ");
```

Comment:
can replace args.split(" ") with args.split(" +"), to ignore multiple consecutive spaces.